Add Tampere University CS Department

ancuongnguyen07 opened the pull request:
# Contributing to CSrankings

Thanks for contributing to CSrankings! Here are some guidelines to getting your pull request accepted.

0. Use a reasonable title that explains what the PR corresponds to (as in, not "Update csrankings-x.csv"). **IF YOU DO NOT DO THIS, YOUR COMMIT WILL BE SUMMARILY REJECTED.**

1. _Do not use Excel to edit any .csv files; Excel incorrectly tries to
convert some Google Scholar entries to formulas, corrupting the
database. Use a text editor like emacs or NotePad instead._

2. _Insert new faculty **in alphabetical order**, not at the end of `csrankings-[a-z].csv`._ **Do not modify `csrankings.csv`, which is auto-generated.** **DO NOT MODIFY `csrankings-[0-9].csv`, WHICH ARE BEING DEPRECATED. USE `csrankings-[a-z].csv` INSTEAD.**

3. _Read and check **all** the boxes below by filling them in with an X._

**Inclusion criteria**

- [X] Make sure that any faculty you add meet the inclusion
criteria. Eligible faculty include only full-time, tenure-track research
faculty members on a given campus who can *solely* advise PhD students in
Computer Science. Faculty not in a CS department or similar who can
advise PhD students in CS can be included regardless of their home
department. Faculty should also have a 75%+ time appointment (check
`old/industry.csv` for faculty who are now more than 25% in industry).

**Updating an affiliation or home page**

- [X] Update affiliations, home pages, and Google Scholar entries by modifying `csrankings-[a-z].csv`. For the Google Scholar entry, just use the alphanumeric identifier in the middle of the URL. If none is there, put NOSCHOLARPAGE.

**Adding one or more faculty members (including an entire department)**

- [X] If the department is not yet listed in CSrankings, the entire faculty needs to be added (not just one faculty member).

- [X] Enter each faculty member's [DBLP](http://dblp.org) name, home page, and Google Scholar entry (just the alphanumeric identifier, not the whole URL) by modifying `csrankings-[a-z].csv` (**the letters correspond to the first letter of the faculty members' names**; please **do not** use the numeric suffixes, which are being obsoleted); include disambiguation suffixes like 0001 as needed. If the faculty entry is currently ambiguous, please do not include them. Send mail to the DBLP maintainers (dblp@dagstuhl.de) with a few publications by a particular faculty member; also, open an issue so that when the DBLP database is updated, that faculty member's information can be added.

- [X] If DBLP has multiple entries for this person, all of them need to be listed. If an alias is not already present in `dblp-aliases.csv`, add it.

-  [X] If the institution you are adding is not in the US,
update `country-info.csv`.

**(Advanced) Quick contribution via a shallow clone** 

A full clone of the CSrankings repository is almost 2GB.  To contribute a change without creating a full local clone of the CSrankings repo, you can do a _shallow clone_.  To do so, follow these steps:

1. Fork the CSrankings repo.  If you have an existing fork, but it is not up to date with the main repository, this technique may not work.  If necessary, delete and re-create your fork to get it up to date.  (Do **not** delete your existing fork if it has unmerged changes you want to preserve!)
2. Do a shallow clone of your fork: `git clone --depth 1 https://github.com/yourusername/CSrankings`.  This will only download the most recent commit, not the full git history.
3. Make your changes on a branch, push them to your clone, and create a pull request on GitHub as usual.

If you want to do another contribution and some time has passed, perform steps 1-3 again, creating a fresh fork and shallow clone.

@romen
@bbbrumley

ancuongnguyen07 committed:
Add Tampere University CS Department

emeryberger commented on csrankings-a.csv:
```diff
@@ -1652,6 +1653,7 @@ Annabelle McIver,Macquarie University,https://researchers.mq.edu.au/en/persons/a
 Annalisa De Bonis,University of Salerno,http://www.di.unisa.it/professori/debonis,XpwdqF8AAAAJ
 Annalisa Massini,Sapienza University of Rome,https://twiki.di.uniroma1.it/twiki/view/Users/AnnalisaMassini,u2H7ZqsAAAAJ
 Annalu Waller,University of Dundee,https://www.computing.dundee.ac.uk/about/staff/27,x_YEVQoAAAAJ
+Annamaria Mesaros,Tampere University,https://www.tuni.fi/en/annamaria-mesaros,tOvdEZIAAAAJ
```
Research fellow, not a tenure-track professor - please remove.

bbbrumley replied:
Academy Research Fellow and Tenure-Track Professor are not mutually exclusive titles in Finland. This is very common, because the former has funding that comes through the [Academy of Finland](https://www.aka.fi/en/), while the latter does not necessarily have that, rather directly from the university.

How about we replace with this URL instead, that will show Annamaria holds both titles -- Academy Research Fellow, and Assistant Professor (tenure track)?

https://researchportal.tuni.fi/en/persons/annamaria-mesaros

emeryberger replied:
That's fine, please do that.

ancuongnguyen07 replied:
Addressed via commit  7cf758c94e

emeryberger commented on csrankings-m.csv:
```diff
@@ -1907,6 +1909,7 @@ Mikio Yamamoto,University of Tsukuba,http://www.mibel.cs.tsukuba.ac.jp/~myama,NU
 Mikkel Abrahamsen,University of Copenhagen,http://www.diku.dk/Ansatte/forskere/?pure=da/persons/289414,AYGv9T8AAAAJ
 Mikkel Thorup,University of Copenhagen,http://www.diku.dk/~mthorup,UPBuOPIAAAAJ
 Mikko H. Lipasti,University of Wisconsin - Madison,https://pharm.ece.wisc.edu/mikko,7tr3iQkAAAAJ
+Mikko J. Ruohonen,Tampere University,https://www.tuni.fi/en/mikko-ruohonen,ElEhWWYAAAAJ
```
In business school, not CS.

bbbrumley replied:
Indeed. We'll remove this one.

emeryberger commented on csrankings-o.csv:
```diff
@@ -28,6 +28,7 @@ Ohad Shamir,Weizmann Institute of Science,https://www.weizmann.ac.il/pages/searc
 Oisin Mac Aodha,University of Edinburgh,https://homepages.inf.ed.ac.uk/omacaod,IfZBjkUAAAAJ
 Ojaswa Sharma,IIIT Delhi,https://www.iiitd.ac.in/ojaswa,jikZgTsAAAAJ
 Okan Yurduseven,Queen's University Belfast,https://www.qub.ac.uk/schools/eeecs/Connect/Staff/BusinessCard/?name=o.yurduseven,sjHXrh8AAAAJ
+Okko Johannes Räsänen,Tampere University,https://www.tuni.fi/en/okko-rasanen,iMVBRk0AAAAJ
```
Research fellow. Please remove this and similar instances.

bbbrumley replied:
Similarly https://researchportal.tuni.fi/en/persons/okko-r%C3%A4s%C3%A4nen will show Okko holding both titles: Academy Research Fellow and Associate Professor (tenure track).

How about if we just replace the URL?

ancuongnguyen07 replied:
Addressed via commit  7cf758c94e

emeryberger commented on csrankings-p.csv:
```diff
@@ -497,6 +498,7 @@ Peixiang Zhao,Florida State University,http://www.cs.fsu.edu/~zhao,ajRhtAkAAAAJ
 Peiyi Tang,University of Arkansas - Little Rock,http://www.ualr.edu/pxtang,NOSCHOLARPAGE
 Peizhao Hu,Rochester Institute of Technology,https://www.cs.rit.edu/~ph,w4PoXGMAAAAJ
 Pejman Lotfi-Kamran,IPM,http://cs.ipm.ac.ir/~plotfi,UZIKgasAAAAJ
+Pekka Jääskeläinen,Tampere University,https://www.tuni.fi/en/pekka-jaaskelainen,wQHcOU0AAAAJ
```
Dead home page.

bbbrumley replied:
Thank you! We will replace it with

https://researchportal.tuni.fi/en/persons/pekka-j%C3%A4%C3%A4skel%C3%A4inen

ancuongnguyen07 replied:
Addressed via commit 8acb67f0e8

emeryberger commented on csrankings-r.csv:
```diff
@@ -892,6 +892,7 @@ Robert P. Burton,Brigham Young University,https://cs.byu.edu/faculty/rpburton,0N
 Robert P. Dick,University of Michigan,https://ece.engin.umich.edu/personnel/dick-robert,p2I_CecAAAAJ
 Robert Pastel,Michigan Technological University,https://www.mtu.edu/cs/department/faculty-staff/faculty/pastel,NOSCHOLARPAGE
 Robert Patro,University of Maryland - College Park,https://www.cs.umd.edu/people/nomad,H36hOqEAAAAJ
+Robert Piché,Tampere University,https://www.tuni.fi/en/robert-piche,Z8UrQ0EAAAAJ
```
Teaching faculty and visitors are not eligible for inclusion.

bbbrumley replied:
Indeed -- we'll remove this one.

Thank you for reviewing the PR! We'll get these changes in next week.

ancuongnguyen07 replied:
Addressed via commit 136e8e1aa2

ancuongnguyen07 committed:
fixup! Add Tampere University CS Department

Update homepages which show both titles: Academy Research Fellow
and Associate Professor (tenure track).

ancuongnguyen07 committed:
fixup! Add Tampere University CS Department

Replace dead homepage

ancuongnguyen07 committed:
fixup! Add Tampere University CS Department

Remove teaching faculty and visitors

ancuongnguyen07 commented:
Thank you for the feedback, I should have addressed all requests via:
- 7cf758c9
- 8acb67f0
- 136e8e1
-  7e449d5d

ancuongnguyen07 committed:
fixup! Add Tampere University CS Department

Remove non-CS faculty
